test/onnx/exporter/test_compat.py

```diff
@@ -86,6 +107,577 @@ def test__unflatten_dynamic_shapes_with_inputs_tree_succeeds_on_tuple_of_mixed_s
+                                0: torch.export.Dim("batch_size"),
+                                2: torch.export.Dim("past_sequence_length"),
+                            },
+                            {
+                                0: torch.export.Dim("batch_size"),
+                                2: torch.export.Dim("past_sequence_length"),
+                            },
+                        ),
+                        (
+                            {
+                                0: torch.export.Dim("batch_size"),
+                                2: torch.export.Dim("past_sequence_length"),
+                            },
+                            {
+                                0: torch.export.Dim("batch_size"),
+                                2: torch.export.Dim("past_sequence_length"),
+                            },
+                        ),
+                        (
+                            {
+                                0: torch.export.Dim("batch_size"),
+                                2: torch.export.Dim("past_sequence_length"),
+                            },
+                            {
+                                0: torch.export.Dim("batch_size"),
+                                2: torch.export.Dim("past_sequence_length"),
+                            },
+                        ),
+                        (
+                            {
+                                0: torch.export.Dim("batch_size"),
+                                2: torch.export.Dim("past_sequence_length"),
+                            },
+                            {
+                                0: torch.export.Dim("batch_size"),
+                                2: torch.export.Dim("past_sequence_length"),
+                            },
+                        ),
+                        (
+                            {
+                                0: torch.export.Dim("batch_size"),
+                                2: torch.export.Dim("past_sequence_length"),
+                            },
+                            {
+                                0: torch.export.Dim("batch_size"),
+                                2: torch.export.Dim("past_sequence_length"),
+                            },
+                        ),
+                    ],
+                ],
+            )
+        ],
+    )
+    def test__from_dynamic_axes_to_dynamic_shapes_succeeds_on_llm(
+        self,
+        model,
+        args,
+        kwargs,
+        input_names,
+        output_names,
+        dynamic_axes,
+        expected_dynamic_shapes,
+    ):
+        dynamic_shapes = _compat._from_dynamic_axes_to_dynamic_shapes(
+            model,
+            args,
+            kwargs,
+            input_names=input_names,
+            output_names=output_names,
+            dynamic_axes=dynamic_axes,
+        )
+
+        # NOTE: torch.export.Dim being an object makes it impossible to compare the objects directly.
```

Comment:
Should we implement `__eq__` on Dim? Just a thought

Reply:
I think it's a valid request? https://github.com/pytorch/pytorch/blob/33191bb664fef338d94586a05bf1f14d17a00340/torch/export/dynamic_shapes.py#L56